src/librustc/hir/mod.rs

```diff
@@ -585,6 +586,8 @@ pub enum PatKind {
     Lit(P<Expr>),
     /// A range pattern, e.g. `1...2`
     Range(P<Expr>, P<Expr>),
+    /// A range pattern excluding the end, e.g. `1..2`
+    ExclusiveRange(P<Expr>, P<Expr>),
```

Comment:
In most cases inclusive and exclusive ranges behave identically, it's probably more convenient to merge them into one variant `Range(P<Expr>, P<Expr>, RangeKind)`.